src/Tracking/Events/ActivatedEvents.php

```diff
@@ -74,5 +85,24 @@ public function maybe_track_activation_source(): void {
 		}
 
 		$this->record_event( 'activated_from_source', $available_source_params );
+
+	}
+
+	/**
+	 * Activate the service.
+	 *
+	 * @return void
+	 */
+	public function activate(): void {
+		$this->maybe_track_activation_source();
```

Comment:
Moving this to activate is a good thing because we can ensure that all the dependencies which are required for this class are available. However if we look at the code of `maybe_track_activation_source` it relies on server variables which are set during the call to any functions hooked into `register_activation_hook`. If we are activating the plugin at a later stage those variables won't be set so it can't detect how the plugin was activated. Note, that this is only for tracking purposes, so it won't break anything, but it will mean we will be tracking activation source incorrectly for any site that activated itself later.

Reply:
I am looking at https://github.com/woocommerce/google-listings-and-ads/blob/develop/src/Tracking/Events/ActivatedEvents.php#:~:text=maybe_track_activation_source , it seems we’re only tracking on install (plugin-install.php), so even currently, we don't track later stage activation at all.

Reply:
Thanks. This has been addressed in the newer commits